### PR TITLE
*: check errors part 2

### PR DESCRIPTION
**Pull request by coocood**
The second part error check, not finished yet.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zz-jason PTAL

File: util/types/datum.go [hunk 1]
```diff
@@ -1408,8 +1423,8 @@ func (d *Datum) ToFloat64(sc *variable.StatementContext) (float64, error) {
 		f, err := d.GetMysqlTime().ToNumber().ToFloat64()
 		return f, err
 	case KindMysqlDuration:
-		f, _ := d.GetMysqlDuration().ToNumber().ToFloat64()
-		return f, nil
+		f, err := d.GetMysqlDuration().ToNumber().ToFloat64()
+		return f, err
```
Comment (shenli) [inline util/types/datum.go]: Should we trace this err?

File: util/types/json/serdes.go [hunk 2]
```diff
@@ -224,8 +232,10 @@ func encodeJSONString(s string, buffer *bytes.Buffer) {
 	byteArray := hack.Slice(s)
 	var varIntBuf = make([]byte, 9)
 	var varIntLen = binary.PutUvarint(varIntBuf, uint64(len(byteArray)))
-	buffer.Write(varIntBuf[0:varIntLen])
-	buffer.Write(byteArray)
+	_, err := buffer.Write(varIntBuf[0:varIntLen])
+	terror.Log(err)
```
Comment (shenli) [inline util/types/json/serdes.go]: Why not return error here?

Comment (coocood) [reply util/types/json/serdes.go]: The function doesn't have an error return value.
Actually, the buffer never returns any error.
I could change all the functions to return an error, but that changes much more code, so I just log the error for the errors that we are sure never happens for now.

Later we can decide if it's worth to add the error return value.

File: util/types/datum.go (hunk 1 above)
Comment (coocood) [reply util/types/datum.go]: I didn't think too much, maybe the traces are intentionally avoided, so just keep it the same as other cases.

Maybe we can do it in another PR to add missing traces.

**Comment by shenli** [comment]
LGTM

File: util/types/mydecimal.go [hunk 3]
```diff
@@ -14,6 +14,8 @@
 package types
 
 import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/terror"
```
Comment (lamxTyler) [inline util/types/mydecimal.go]: Put these downward?

**Comment by zz-jason** [comment]
to #4668

**Comment by coocood** [comment]
@lamxTyler PTAL

**Comment by lamxTyler** [comment]
LGTM

**Pull request merged**